app/controllers/submission_similarities_controller.rb

```diff
@@ -29,16 +30,77 @@ class SubmissionSimilaritiesController < ApplicationController
       controller.send :authenticate_actions_for_role, UserCourseMembership::ROLE_STUDENT,
                                                       course: @course,
                                                       only: [ ]
+      controller.send :authenticate_actions_for_role, UserCourseMembership::ROLE_GUEST,
+                                                      course: @course,
+                                                      only: [ :index, :show ]
     end
   }
 
   # GET /assignments/1/submission_similarities
   def index
+    # obtain parameters (if any)
+    @hashed_url = session[:hashed_url]
+    @displayDialog = session[:displayDialog]
+    session[:displayDialog] = false
+ 
     @submission_similarities = SubmissionSimilarity.where(
       assignment_id: @assignment.id 
     ).order('similarity desc').paginate(page: params[:page], per_page: 20)
   end
 
+  def create_guest_user
+    @user = User.find_by_id(session[:user_id]) 
+    @guest_user = GuestUsersDetail.find_by_user_id(@user.id)
+    # do not allow to create a sharable link if user is a guest user
+    if (@guest_user)
+      redirect_to assignment_submission_similarities_url(params[:id]), notice: 'You cannot create a sharable link as you are using SSID as a guest user. To create a sharable link and/or use other features, log in to SSID.'
```

Comment:
Global replace: sharable => shareable (in all files in the commit).

Reply:
Thanks for spotting this as well. I have changed all these spelling errors.